tests/test_efadv.py

```diff
@@ -0,0 +1,34 @@
+# SPDX-License-Identifier: (GPL-2.0 OR Linux-OpenIB)
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All rights reserved.
+"""
+Test module for efa direct-verbs.
+"""
+
+import argparse
+import pyverbs.device as d
+import pyverbs.providers.efa.efadv as efa
+import sys
+import unittest
+
+from tests.base import PyverbsAPITestCase
+
+
+class EfaDvTest(PyverbsAPITestCase):
+    """
+    Test various functionalities of the direct verbs class.
+    """
+    def test_efadv_query(self):
+        """
+        Verify that it's possible to read EFA direct-verbs.
+        """
+        for ctx, attr, attr_ex in self.devices:
+            with efa.EfaContext(name=ctx.name) as efa_ctx:
+                try:
+                    efa_attrs = efa_ctx.query_efa_device()
+                    if not efa_attrs:
+                        raise PyverbsError('efadv_query_device failed')
+                except:
+                    # Triggered for non-efa devices.
```

Comment:
This will show that the test "passed" even on other devices. I expect this test to be skipped on non-efa devices. (by raising unittest.SkipTest, you can check it in other tests).
You should do something like:

```
try:
                    efa_attrs = efa_ctx.query_efa_device()
                    if not efa_attrs:
                        raise PyverbsError('efadv_query_device failed')
                except PyverbsRDMAError as ex:
                        if ex.error_code == errno.EOPNOTSUPP:
                                    raise unittest.SkipTest('relevant msg (maybe Not a EFA device)')
                         else:
                                   raise ex
```

Reply:
Done